loopchain/blockchain/blocks/v0_4/block_builder.py

```diff
@@ -0,0 +1,281 @@
+import time
+from functools import reduce
+from operator import or_
+from typing import List
+
+from loopchain.blockchain.types import ExternalAddress, Hash32, BloomFilter
+from loopchain.blockchain.transactions import TransactionVersioner
+from loopchain.blockchain.blocks import Block, BlockBuilder as BaseBlockBuilder, BlockProverType
+from loopchain.blockchain.blocks.v0_4 import BlockHeader, BlockBody, BlockProver
+from loopchain.blockchain.blocks.block import NextRepsChangeReason
+from loopchain.blockchain.votes.v0_4 import BlockVote, LeaderVote
+
+
+class BlockBuilder(BaseBlockBuilder):
+    version = BlockHeader.version
+    BlockHeaderClass = BlockHeader
+    BlockBodyClass = BlockBody
+
+    def __init__(self, tx_versioner: 'TransactionVersioner'):
+        super().__init__(tx_versioner)
+
+        # Attributes that must be assigned
+        self.reps: List[ExternalAddress] = None
+        self.next_reps: List[ExternalAddress] = None
+        self.next_reps_hash: Hash32 = None
+        self.next_reps_change_reason: NextRepsChangeReason = NextRepsChangeReason.NoChange
+        self.leader_votes: List[LeaderVote] = []
+        self.prev_votes: List[BlockVote] = None
+
+        # Attributes to be assigned(optional)
+        self.fixed_timestamp: int = None
+        self.state_hash: 'Hash32' = None
+        self.next_leader: 'ExternalAddress' = None
+        self.is_max_made_block_count: bool = None
```

Comment:
Is this a kind of attrs to be assigned in `Block`?
and default value `None` is questionable.

Reply:
above `self.fixed_timestamp` and below `self._timestamp`, `self._receipts` are set `None` as default.
I think @zsaladin will explain better than me.